edxval/views.py

```diff
@@ -162,6 +170,52 @@ def post(self, request):
         return response
 
 
+class VideoStatusView(APIView):
+    """
+    A Video View, used by VEDA to update the status of a video.
+
+    Note:
+        Currently, the valid statuses are `transcription_in_progress` and `transcription_ready` because it was intended
+        to only be used for video transcription process by VEDA but if you found it helpful to your needs, you can add
+        more statuses so that you can use it for updating other video statuses too.
+    """
+    def patch(self, request):
+        """
+        Update the status of a video.
+        """
+        attrs = ('edx_video_id', 'status')
+        missing = [attr for attr in attrs if attr not in request.data]
+        if missing:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data=dict(message=u'{missing} must be specified.'.format(missing=' and '.join(missing)))
+            )
+
+        edx_video_id = request.data['edx_video_id']
+        video_status = request.data['status']
+        if video_status not in VALID_VIDEO_STATUSES:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data=dict(message=u'"{status}" is not a valid Video status.'.format(status=video_status))
+            )
+
+        try:
+            video = Video.objects.get(edx_video_id=edx_video_id)
+            video.status = video_status
+            video.save()
+            response_status = status.HTTP_200_OK
+            response_payload = {}
+        except Video.DoesNotExist:
+            response_status = status.HTTP_400_BAD_REQUEST
+            response_payload = {
+                'message': u'Video is not found for specified edx_video_id: {edx_video_id}'.format(
+                    edx_video_id=edx_video_id
+                )
+            }
+
+        return Response(status=response_status, data=response_payload)
```

Comment:
Let's make it 200 and have another response in except block to return 400 and respective payload, since we're already doing it on many places.